pytest_httpserver/httpserver.py

```diff
@@ -45,6 +48,43 @@ class NoMethodFoundForMatchingHeaderValueError(Error):
     pass
 
 
+class WaitingSettings:
+    """Class for providing default settings and storing them in HTTPServer
+
+    :param raise_assertions: whether raise assertions on unexpected request or timeout or not
+    :param stop_on_nohandler: whether stop on unexpected request or not
+    :param timeout: time (in seconds) until time is out
+    """
+
+    def __init__(self, raise_assertions: bool = True, stop_on_nohandler: bool = True, timeout: float = 5):
+        self.raise_assertions = raise_assertions
+        self.stop_on_nohandler = stop_on_nohandler
+        self.timeout = timeout
+
+
+class Waiting:
+    """Class for HTTPServer.wait context manager
+
+    This class should not be instantiated directly."""
+
+    def __init__(self):
+        self._result = None
+        self.start = datetime.now()
```

Comment:
Could this be utcnow()? It would be great to avoid DST changes.

You may also want to use monotonic time if you are interested as that is designed explicitly to measure time between events.